odoo/addons/base/models/res_company.py

```diff
@@ -31,8 +31,6 @@ def _get_user_currency(self):
 
     name = fields.Char(related='partner_id.name', string='Company Name', required=True, store=True, readonly=False)
     sequence = fields.Integer(help='Used to order Companies in the company switcher', default=10)
-    parent_id = fields.Many2one('res.company', string='Parent Company', index=True)
-    child_ids = fields.One2many('res.company', 'parent_id', string='Child Companies')
```

Comment:
We should keep these fields to allow using them for other purposes, or for specific record rules. It will also require less changes elsewhere.